planner/core/logical_plans.go

```diff
@@ -835,6 +835,10 @@ func (ds *DataSource) fillIndexPath(path *util.AccessPath, conds []expression.Ex
 			if !alreadyHandle {
 				path.IdxCols = append(path.IdxCols, handleCol)
 				path.IdxColLens = append(path.IdxColLens, types.UnspecifiedLength)
+				// Also updates the map that maps the index id to its prefix column ids.
+				if len(ds.tableStats.HistColl.Idx2ColumnIDs[path.Index.ID]) == len(path.Index.Columns) {
+					ds.tableStats.HistColl.Idx2ColumnIDs[path.Index.ID] = append(ds.tableStats.HistColl.Idx2ColumnIDs[path.Index.ID], handleCol.UniqueID)
```

Comment:
I think this would affect many other uses of `Idx2ColumnIDs`, like `getGroupNDVs()`.

Reply:
The GroupNDV checks that `len(Idx2ColumnIDs[idxID]) == len(idxInfo.Columns)`